Show pending shares menu only if feature enabled

printminion-co opened the pull request:
## Summary
Currently the "pending shares" is always available if feature enabled or not

Documentation: https://docs.nextcloud.com/server/latest/admin_manual/configuration_server/config_sample_php_parameters.html

![Selection_20250612-003](https://github.com/user-attachments/assets/6c21c3e2-e610-4a1a-9dbc-606d7fd59239)


## Proposal
With this change the "pending shares" will be displayed only if feature is enabled.

### "pending shares" menu item is shown on
./occ config:system:set --value true --type boolean  -- sharing.enable_share_accept

### "pending shares" menu item is removed on
./occ config:system:set --value false --type boolean  -- sharing.enable_share_accept

### run test via 
```script
 phpunit --bootstrap tests/bootstrap.php apps/files_sharing/tests/Listener/LoadAdditionalListenerTest.php
```


<!--
  - 🚨 SECURITY INFO
  -
  - Before sending a pull request that fixes a security issue please report it via our HackerOne page (https://hackerone.com/nextcloud) following our security policy (https://nextcloud.com/security/). This allows us to coordinate the fix and release without potentially exposing all Nextcloud servers and users in the meantime.
-->


## TODO

- [ ] ...

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

printminion-co committed:
feat(files-sharing): show pending shares menu only if feature enabled

# menu is removed on
./occ config:system:set --value true --type boolean  -- sharing.enable_share_accept

# menu is shown on
./occ config:system:set --value false --type boolean  -- sharing.enable_share_accept

Signed-off-by: Misha M.-Kupriyanov <kupriyanov@strato.de>

come-nc commented on apps/files_sharing/lib/Listener/LoadAdditionalListener.php:
```diff
@@ -31,5 +33,14 @@ public function handle(Event $event): void {
+		$this->provideInitialStates();
+	}
+
+	private function provideInitialStates(): void {
+		$initialState = Server::get(InitialStateService::class);
+		$config = Server::get(IConfig::class);
+		$defaultAcceptSystemConfig = $config->getSystemValueBool('sharing.enable_share_accept');
+		$initialState->provideInitialState(Application::APP_ID, 'accept_default', $defaultAcceptSystemConfig);
```
As documented in config sample, this can be overridden by any user:
```php
/**
 * Set to true to enable that internal shares need to be accepted by the users by default.
 * Users can change this for their account in their personal sharing settings
 */
'sharing.enable_share_accept' => false,
```

Also, users may have pending shares from before the option was set, I suppose?
I also fear that internal share are not the only kind and that users still need to accept pending federated shares even if the option is set, at least that’s how I understand the comment in config sample.

artonge reviewed (commented):
Could we instead check if there are any pending shares, and hide the menu if there are none?

artonge commented on apps/files_sharing/lib/Listener/LoadAdditionalListener.php:
```diff
@@ -33,5 +38,39 @@ public function handle(Event $event): void {
+	}
+
+	private function provideInitialStates(): void {
+		$userSession = Server::get(IUserSession::class);
```
Is it not possible to inject the dependency in the constructor?

artonge replied:
Same for the other services bellow.

artonge commented on apps/files_sharing/lib/Listener/LoadAdditionalListener.php:
```diff
@@ -33,5 +38,39 @@ public function handle(Event $event): void {
+		$externalManager = Server::get(ExternalManager::class);
+
+		$acceptDefault = $config->getSystemValueBool('sharing.enable_share_accept');
+		$initialState->provideInitialState(Application::APP_ID, 'accept_default', $acceptDefault);
```
Let's use the same naming scheme:

```suggestion
		$initialState->provideInitialState(Application::APP_ID, 'sharing.enable_share_accept', $acceptDefault);
```

printminion-co committed:
refactor(files-sharing): inject IManager into LoadAdditionalListener

Resolve the share manager through constructor injection instead of
Server::get() so the listener follows the app's dependency-injection
conventions.

Signed-off-by: Misha M.-Kupriyanov <kupriyanov@strato.de>

printminion-co committed:
feat(files-sharing): show pending shares menu only if feature enabled

# menu is removed on
./occ config:system:set --value true --type boolean  -- sharing.enable_share_accept

# menu is shown on
./occ config:system:set --value false --type boolean  -- sharing.enable_share_accept

Signed-off-by: Misha M.-Kupriyanov <kupriyanov@strato.de>

printminion-co committed:
chore(files-sharing): rebuild assets

Signed-off-by: Misha M.-Kupriyanov <kupriyanov@strato.de>
